packages/perf-benchmarks/scripts/build.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+/**
+ * Builds the HTML and tachometer.json files necessary to run the benchmarks.
+ */
+
+const path = require('path');
+const fs = require('fs');
+const { promisify } = require('util');
+const glob = promisify(require('glob'));
+
+const writeFile = promisify(fs.writeFile);
+
+// lwc packages that need to be swapped in when comparing the current code to the latest tip-of-tree code.
+const swappablePackages = [
+    '@lwc/engine-dom',
+    '@lwc/engine-server',
+    '@lwc/synthetic-shadow',
+    'perf-benchmarks-components',
+];
+
+function createHtml(benchmarkFile) {
+    return `
+    <!doctype html>
+    <html>
+      <head>
+        <title>Benchmark</title>
+      </head>
+      <body>
+        <script type="module">
+          // Always test performance in production mode.
+          // Note this is needed for modules like @lwc/synthetic-shadow, since we can't run it through
+          // @rollup/plugin-replace since Tachometer will serve it.
+          window.process = { env: { NODE_ENV: 'production' } };
+        </script>
+        <script type="module" src="./${path.basename(benchmarkFile)}"></script>
+      </body>
+    </html>
+  `.trim();
+}
+
+function createTachometerJson(htmlFilename, benchmarkName) {
+    return {
+        $schema: 'https://raw.githubusercontent.com/Polymer/tachometer/master/config.schema.json',
+        sampleSize: 50, // minimum number of samples to run
+        timeout: 0, // timeout in minutes during auto-sampling (after the first minimum samples). If 0, no auto-sampling
+        benchmarks: [
+            {
+                url: htmlFilename,
+                name: benchmarkName,
+                browser: {
+                    name: 'chrome',
+                    headless: true,
+                },
+                measurement: {
+                    mode: 'performance',
+                    entryName: 'benchmark-run',
+                },
+                expand: [
+                    {
+                        name: `${benchmarkName}-this-change`,
+                    },
+                    {
+                        name: `${benchmarkName}-tip-of-tree`,
+                        packageVersions: {
+                            label: 'tip-of-tree',
+                            dependencies: Object.fromEntries(
+                                swappablePackages.map((pkg) => [
+                                    pkg,
+                                    {
+                                        kind: 'git',
+                                        repo: 'https://github.com/salesforce/lwc.git',
+                                        ref: 'master',
```

Comment:
If you want to actually test this PR locally, you'll have to change `master` to `nolan/tachometer` here.

Reply:
should we make this configurable? like an env var? For example, if this were a CI job, it must compare to the base branch of the PR.

Reply:
In CI it would still want to compare to master, no? It's comparing whatever code is in the PR to whatever is in master.

But yes, making it configurable would definitely make it easier to play around with it locally, e.g. to test against previous releases. I think we can add env vars for this.

Reply:
Done!